front/cypress/e2e/input_form_builder/create_and_fill_form.cy.ts

```diff
@@ -0,0 +1,173 @@
+import { randomString } from '../../support/commands';
+
+describe('Input form builder', () => {
+  const projectTitle = randomString();
+  const projectDescription = randomString();
+  let questionTitle = randomString();
+  const answer = randomString();
+  const ideaTitle = randomString(40);
+  const ideaContent = randomString(60);
+  const projectDescriptionPreview = randomString(30);
+  let projectId: string;
+  let projectSlug: string;
+
+  beforeEach(() => {
+    if (projectId) {
+      cy.apiRemoveProject(projectId);
+    }
+
+    cy.setAdminLoginCookie();
+    questionTitle = randomString();
+
+    cy.apiCreateProject({
+      type: 'continuous',
+      title: projectTitle,
+      descriptionPreview: projectDescriptionPreview,
+      description: projectDescription,
+      publicationStatus: 'published',
+      participationMethod: 'ideation',
+    }).then((project) => {
+      projectId = project.body.data.id;
+      projectSlug = project.body.data.attributes.slug;
+    });
+  });
+
+  it('has a working idea form with all the defaults', () => {
+    cy.visit(`admin/projects/${projectId}/ideaform`);
+    cy.get('[data-cy="e2e-edit-input-form"]').click();
+
+    cy.get('form').submit();
+    // Should show success message on saving
+    cy.get('[data-testid="feedbackSuccessMessage"]').should('exist');
+
+    // Fill in the form
+    cy.visit(`/projects/${projectSlug}/ideas/new`);
+    cy.acceptCookies();
+
+    cy.get('#e2e-idea-new-page');
+    cy.get('#idea-form');
+    cy.contains('Add new idea').should('exist');
+    // add a title and description
+    cy.get('#e2e-idea-title-input input').type(ideaTitle);
+    cy.get('#e2e-idea-description-input .ql-editor').type(ideaContent);
+
+    // verify the title and description
+    cy.get('#e2e-idea-title-input input').should('contain.value', ideaTitle);
+    cy.get('#e2e-idea-description-input .ql-editor').contains(ideaContent);
+
+    // add a topic
+    cy.get('.e2e-topics-picker').find('button').eq(4).click();
+
+    // verify that the topic has been selected
+    cy.get('.e2e-topics-picker')
+      .find('button.selected')
+      .should('have.length', 1);
+
+    // add a location
+    cy.get('.e2e-idea-form-location-input-field input').type(
+      'Boulevard Anspach Brussels{enter}'
+    );
+    cy.get(
+      '.e2e-idea-form-location-input-field #PlacesAutocomplete__autocomplete-container div'
+    )
+      .first()
+      .click();
+    cy.wait(500);
```

Comment:
Do you think this could be flaky when run on CI? Sometimes those runs are really slow :thinking:

Reply:
Fixed